src/twisted/conch/scripts/ckeygen.py

```diff
@@ -202,45 +205,74 @@ def _defaultPrivateKeySubtype(keyType):
         return "PEM"
 
 
-def printFingerprint(options):
-    if not options["filename"]:
-        filename = os.path.expanduser("~/.ssh/id_rsa")
-        options["filename"] = input("Enter file in which the key is (%s): " % filename)
-    if os.path.exists(options["filename"] + ".pub"):
-        options["filename"] += ".pub"
+def _getKeyOrDefault(
+    options: Dict[Any, Any],
+    inputCollector: Optional[Callable] = None,
+    keyTypeName: str = "rsa",
+) -> str:
+    """
+    If C{options["filename"]} is None, prompt the user to enter a path
+    or attempt to set it to .ssh/id_rsa
+    @param options: command line options
+    @type options: L{Dict}
```

Comment:
no need to add `@type` here, as long as we have type hinting.

and I think is best not to have `@type` in the docstring to avoid future conflicts.

Reply:
Updated.